common/sgx/internal.edl

```diff
@@ -11,17 +11,126 @@
 **==============================================================================
 */
 
-enclave {
+enclave
+{
+    include "openenclave/bits/types.h"
+    include "openenclave/internal/sgxtypes.h"
 
-    trusted {
+    trusted
+    {
+        public uint32_t oe_get_sgx_report_ecall(
+            [in, size=opt_params_size] const void* opt_params,
+            size_t opt_params_size,
+            [out] sgx_report_t* report);
 
-        /* This function returns oe_internal_ping_ocall(value). */
-        public int oe_internal_ping_ecall(int value);
+        public uint32_t oe_verify_report_ecall(
+            [in, size=report_size] const void* report,
+            size_t report_size);
+
+        public uint32_t oe_get_public_key_ecall(
+            [in] const oe_asymmetric_key_params_t* key_params,
+            [in, size=key_info_size] const void* key_info,
+            size_t key_info_size,
+            [out, size=key_buffer_size] void* key_buffer,
+            size_t key_buffer_size,
+            [out ] size_t* key_buffer_size_out);
+
+        public uint32_t oe_get_public_key_by_policy_ecall(
+            uint32_t seal_policy,
+            [in] const oe_asymmetric_key_params_t* key_params,
+            [out, size=key_buffer_size] void* key_buffer,
+            size_t key_buffer_size,
+            [out ] size_t* key_buffer_size_out,
+            [out, size=key_info_size] void* key_info,
+            size_t key_info_size,
+            [out ] size_t* key_info_size_out);
+
+        public void oe_log_init_ecall(
+            [in, string] const char* enclave_path,
+            uint32_t log_level);
     };
 
-    untrusted {
+    untrusted
+    {
+        uint32_t oe_get_qetarget_info_ocall(
+            [out] sgx_target_info_t* target_info);
+
+        uint32_t oe_get_quote_ocall(
+            [in] const sgx_report_t* sgx_report,
+            [out, size=quote_size] void* quote,
+            size_t quote_size,
+            [out] size_t* quote_size_out);
+
+        uint32_t oe_get_qe_identify_info_ocall(
+            [out, size=qe_id_info_size] void* qe_id_info,
+            size_t qe_id_info_size,
+            [out] size_t* qe_id_info_size_out,
+            [out, size=issuer_chain_size] void* issuer_chain,
+            size_t issuer_chain_size,
+            [out] size_t* issuer_chain_size_out);
+
+        uint32_t oe_get_revocation_info_ocall(
```

Comment:
No, arrays of pointers are not planned for deep copy. I mentioned this to Andy and he said it would be too hard to support. I'm planning a document on my experience with EDL.